exporter/prometheusexporter/end_to_end_test.go

```diff
@@ -51,6 +51,10 @@ func TestEndToEndSummarySupport(t *testing.T) {
 			// Serve back the metrics as if they were from DropWizard.
 			_, err := rw.Write([]byte(dropWizardResponse))
 			require.NoError(t, err)
+		default:
+			// Non-blocking serve for DropWizard
```

Comment:
If you're going to do this, just remove `waitForScrape` entirely since this means it will never block.  I would be reluctant to do so, however, without fully understanding why it exists and what results from removing it.

Reply:
Removing `waitForScrapes` entirely will also remove the guarantee that Prometheus Receiver scrapes. Can I remove the `waitForScrapes`, and introduce `time.Sleep(16 *time.Seconds) `at Line 120 to ensure that Prometheus Receiver scrapes from test server atleast 8 times.